### ISSUE TITLE
[Bug] [Jac Cloud] jid() call overrides core jaclang outside of jac serve

### ISSUE THREAD

**Issue by ChrisIsKing** [Jaseci-Labs/jaseci#1339]
## **Describe the bug**

When running a Jac program with `jac-cloud` installed, the built-in `jid()` function is overridden, resulting in an error. The issue occurs when the program is executed with `jac run` rather than `jac serve`. The expected behavior is that built-in functions like `jid()` should only be overridden by jac-cloud when using `jac serve`, not during normal execution of a Jac program.

### **To Reproduce**
1. Install `jac-cloud` in your Python environment.
2. Use the following Jac program:

```
node test_node {
    has value: int = 0;
}

with entry {
    root ++> (node_ref:=test_node());
    id = jid(node_ref);
    print(id);
    print(node_ref.__jac__.id);
}
```
3. Run the program with `jac run`
4. Observe the following error:

```
ERROR - Error: 'NodeAnchor' object has not attribute 'ref_id'
        raise AttributeError(
        ^
    at __getattr__() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/runtimelib/architype.py:235
    at object_ref() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jac_cloud/plugin/jaseci.py:454
    at _multicall() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_callers.py:103
    at _multicall() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_callers.py:139
    at _hookexec() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_manager.py:120
    at __call__() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_hooks.py:513
    at object_ref() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/plugin/feature.py:57
    at jid() /Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/plugin/builtin.py:46
    at <module> /Users/chrisisking/dev/jaclang_sandbox/jid.jac:7
Traceback (most recent call last):
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/bin/jac", line 8, in <module>
    sys.exit(start_cli())
             ^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/cli/cli.py", line 514, in start_cli
    ret = command.call(**args_dict)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/cli/cmdreg.py", line 25, in call
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/cli/cli.py", line 91, in run
    jac_import(
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/plugin/feature.py", line 119, in jac_import
    return pm.hook.jac_import(
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/plugin/default.py", line 268, in jac_import
    import_result = JacImporter(JacMachine.get()).run_import(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/runtimelib/importer.py", line 354, in run_import
    raise e
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/runtimelib/importer.py", line 351, in run_import
    exec(codeobj, module.__dict__)
  File "/Users/chrisisking/dev/jaclang_sandbox/jid.jac", line 7, in <module>
    id = jid(node_ref);
          ^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/plugin/builtin.py", line 46, in jid
    return Jac.object_ref(obj)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/plugin/feature.py", line 57, in object_ref
    return pm.hook.object_ref(obj=obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/vendor/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jac_cloud/plugin/jaseci.py", line 454, in object_ref
    return str(obj.__jac__.ref_id)
               ^^^^^^^^^^^^^^^^^^
  File "/Users/chrisisking/dev/jaclang_sandbox/venv/lib/python3.12/site-packages/jaclang/runtimelib/architype.py", line 235, in __getattr__
    raise AttributeError(
AttributeError: 'NodeAnchor' object has not attribute 'ref_id'
```
5. Uninstall `jac-cloud` from the Python environment.
6. Run the same program again using `jac run` and observe that it works without error.

**Comment by amadolid** [comment]
This should be working on latest version. Please let me know if issue still persist

**Comment by ChrisIsKing** [comment]
Closing issue. Fixed in #1316.
